src/library/dlgtagfetcher.cpp

```diff
@@ -245,45 +275,56 @@ void DlgTagFetcher::fetchTagFinished(
 }
 
 void DlgTagFetcher::slotNetworkResult(
-        int httpError, const QString& app, const QString& message, int code) {
+        int httpError,
+        const QString& app,
+        const QString& message,
+        int code) {
     m_networkResult = httpError == 0 ? NetworkResult::UnknownError : NetworkResult::HttpError;
     m_data.m_pending = false;
-    QString strError = tr("HTTP Status: %1");
-    QString strCode = tr("Code: %1");
-    httpStatus->setText(strError.arg(httpError) + "\n" + strCode.arg(code) + "\n" + message);
-    QString unknownError = tr("Mixxx can't connect to %1 for an unknown reason.");
-    cantConnectMessage->setText(unknownError.arg(app));
-    QString cantConnect = tr("Mixxx can't connect to %1.");
-    cantConnectHttp->setText(cantConnect.arg(app));
+    if (m_networkResult == NetworkResult::UnknownError) {
+        QString unknownError = tr("Mixxx can't connect to %1 for an unknown reason.");
+        loadingProgressBar->setFormat(unknownError.arg(app));
+    } else if (m_networkResult == NetworkResult::HttpError) {
+        QString cantConnect = tr("Mixxx can't connect to %1.");
+        loadingProgressBar->setFormat(cantConnect.arg(app));
+    } else {
+        QString strError = tr("HTTP Status: %1");
+        QString strCode = tr("Code: %1");
+        loadingProgressBar->setFormat(strError.arg(httpError) + " " +
+                strCode.arg(code) + " " + message);
+    }
     updateStack();
 }
 
 void DlgTagFetcher::updateStack() {
+    m_progressBarStep = 0;
+    loadingProgressBar->setValue(m_progressBarStep);
+    results->clear();
+    addDivider(tr("Original tags"), results);
+    addTrack(trackColumnValues(*m_track), -1, results);
     if (m_data.m_pending) {
-        stack->setCurrentWidget(loading_page);
         return;
     } else if (m_networkResult == NetworkResult::HttpError) {
-        stack->setCurrentWidget(networkError_page);
+        loadingProgressBar->setValue(loadingProgressBar->maximum());
         return;
     } else if (m_networkResult == NetworkResult::UnknownError) {
-        stack->setCurrentWidget(generalnetworkError_page);
+        loadingProgressBar->setValue(loadingProgressBar->maximum());
         return;
     } else if (m_data.m_results.isEmpty()) {
-        stack->setCurrentWidget(error_page);
+        loadingProgressBar->setValue(loadingProgressBar->maximum());
+        QString emptyMessage = tr("Mixxx could not find this track in the musicbrainz database.");
+        loadingProgressBar->setFormat(emptyMessage);
         return;
     }
     btnApply->setEnabled(true);
-    stack->setCurrentWidget(results_page);
-
-    results->clear();
+    loadingProgressBar->setValue(loadingProgressBar->maximum());
+    QString finishedMessage = tr("Mixxx found the metadata.");
```

Comment:
Mabye:
```suggestion
    QString finishedMessage = tr("Results ready to select and apply.");
```
or such ..

Reply:
Done :+1: